### ISSUE TITLE
Do not update the minimap on every keystroke

### ISSUE THREAD

**Issue by robertrossmann** [atom-minimap/minimap#580]
# Proposal

I propose that the minimap package is modified so that the minimap's contents are updated at certain periods instead of on every keystroke or other similar event. This should dramatically increase performance when working with large files.

# Rationale

I just realised that when working with large JS files (>3k SLOC) each keystroke takes approximately 100ms to appear on the screen (this translates to about 10fps when constantly writing).

As proof, I am attaching screenshot from Chrome's Timeline profiler. You can clearly see the framerates mentioned and where most of the time is spent.

![keystrokes - minimap](https://cloud.githubusercontent.com/assets/3058150/24208667/77a92192-0f24-11e7-9eb5-8e31d1de8a39.png)

# Steps to reproduce

- Start a clean installation of Atom (you can `mv ~/.atom/packages ~/.atom/packages.disabled` to get a clean Atom installation)
- Install atom-minimap package
- Open up a JavaScript file with more than 2k SLOC
- Start typing
- Despair 😱

---

Causing a redraw on each keystroke seems like a perf killer. On the other hand, redrawing can be greatly optimised by:

- Waiting for a change to occur which would normally cause the minimap to redraw
- Starting up a timer which, when expired, would cause the actual redraw to occur
- Moving the timer forward in time when a redraw event is triggered while the timer is still running

**In other words, only redraw the minimap once the user has stopped typing.** :)

I have not looked at the source code to estimate how difficult this would be to implement, I just wanted to point out a serious performance bottleneck.

I would love to provide further information if you need it. Thank you for considering this proposal.

> PS. If there is another way to eliminate this slowness I'm all game. This proposal assumes that the code, as currently written, cannot be optimised further.
